var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -1094,33 +1094,26 @@ def cache_test_sources(self):
         install test subdirectory for use during `spack test run`."""
         # Clean the 'examples' directory -- at least one example is always built
         # and we do not want to cache executables.
-        make("examples/clean", parallel=False)
+        make(join_path("examples", "clean"), parallel=False)
         self.cache_extra_test_sources([self.examples_src_dir, self.examples_data_dir])
 
-    def test(self):
-        test_dir = join_path(self.test_suite.current_test_cache_dir, self.examples_src_dir)
-
+    def test_ex10(self):
+        """build and run ex10(p)"""
         # MFEM has many examples to serve as a suitable smoke check. ex10
         # was chosen arbitrarily among the examples that work both with
         # MPI and without it
+        test_dir = join_path(self.test_suite.current_test_cache_dir, self.examples_src_dir)
+
+        makefile = join_path(self.prefix.share.mfem, "config.mk")
```

Comment:
This can use the `config_mk` property defined below.

Reply:
Fixed.